CMake/config/CompilerFlagsHelpers.cmake

```diff
@@ -141,6 +141,9 @@ foreach(COMPILER_LANGUAGE ${SUPPORTED_COMPILER_LANGUAGE_LIST})
 			## Intel O3 is extreme
 			set(CMAKE_${COMPILER_LANGUAGE}_OPT_AGGRESSIVE "-O2")
 		endif()
+		if (CMAKE_${COMPILER_LANGUAGE}_COMPILER_ID STREQUAL "PGI")
+			set(CMAKE_${COMPILER_LANGUAGE}_WARNING_ALL "")
+		endif()
```

Comment:
After looking at this I realise that `-Wall` shouldn't be set in this block...set empty this on line 126 and remove this block.

Reply:
Fixed.

Reply:
Regarding the "Endian" error, If I comment out the line that causes issues the tests are correctly compiled. The test then fail but I'm not sure it's connected to that. Am I the only one that is having that kind of errors while compiling tests?

Reply:
Endian error fixed, created separate pull request.